### ISSUE TITLE
SELFIES encoding bug

### ISSUE THREAD

**Issue by Kohulan** [aspuru-guzik-group/selfies#52]
Dear @MarioKrenn6240 ,

I have an issue with SELFIES encoding. In the following SMILES, when encoded into SELFIES, the SELFIES contains a '[N]' token. In the original SMILES, there are no 'N' atoms. It would be helpful if you could clarify why that is so. When we decode the SELFIES it does decode back to the original SMILES.

SMILES : 'C1C(C(OC2=CC=CC=C21)C3=CC=CC=C3)O'
SELFIES : '[C][C][Branch2_1][Ring1][Branch1_3][C][Branch1_1]**[N]**[O][C][=C][C][=C][C][=C][Ring1][Branch1_2][Ring1][Branch2_3][C][=C][C][=C][C][=C][Ring1][Branch1_2][O]'
Decoded : 'C2C(C(OC1=CC=CC=C12)C3=CC=CC=C3)O'

-Kohulan

**Comment by MarioKrenn6240** [comment]
Hi Kohulan,
What you see is not a bug, but a property of SELFIES. For that, you need to know that lengths of branches and rings in SELFIES are encoded locally. That means, the symbol after `[BranchX_Y]` and `[RingZ]` is _interpreted_ as a number that defines the length of the branch/ring.

in your example, `[N]` is written after a `[Branch1_1]`, so it's interpreted as a number. Now the connection between symbols and numbers is defined [here](https://selfies.readthedocs.io/en/latest/tutorial.html#index-symbols). You see that `[N]` stands for index=10 and thus the length=(index+1)=11.

To sum up, what you see is a property of the local encoding of branches/rings as numbers. I explain this in more detail in this [talk, at around minute 21.00](https://www.youtube.com/watch?v=CaIyUmfGXDk&t=1297s).

**Comment by Kohulan** [comment]
Hi @MarioKrenn6240 ,

Thanks a lot for the clarification. It was visually giving a wrong expression that's why I raised this issue.

-Kohulan
